src/DynamoUtilities/Hash.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Dynamo.Utilities
+{
+    internal class Hash
+    {
+        /// <summary>
+        /// Get the hash value
+        /// </summary>
+        /// <param name="bytes">input as a byte array</param>
+        /// <returns>hash as a byte array</returns>
+        internal static byte[] GetHash(byte[] bytes)
+        {
+            using (var hashAlgorithm = SHA256.Create())
+            {
+                return hashAlgorithm.ComputeHash(bytes);
+            }
+        }
+
+        /// <summary>
+        /// Get the hash value
+        /// </summary>
+        /// <param name="str">input as a string</param>
+        /// <returns>hash as a byte array</returns>
+        internal static byte[] GetHashFromString(string str)
+        {
+            var bytes = Encoding.UTF8.GetBytes(str);
+            return GetHash(bytes);
+        }
+
+        /// <summary>
+        /// Get a valid filename for a hash
+        /// </summary>
+        /// <param name="bytes">hash as a byte array</param>
+        /// <returns>hash a valid filename string</returns>
+        internal static string GetFilenameFromHash(byte[] bytes)
```

Comment:
is the encoding/decoding logic specific to filenames ? looks pretty generic to me

Reply:
What we get pack are guaranteed to work as a file name in all file systems. That's why I named it that way (and used base32).